packages/toolkit/src/lib/react-query-service/organization/useLeaveOrganization.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { leaveOrganizationMutation } from "../../vdp-sdk";
+import type { Nullable } from "../../type";
+
+export const useLeaveOrganization = () => {
+  const queryClient = useQueryClient();
+  return useMutation(
+    async ({
+      organizationName,
+      userName,
+      accessToken,
+    }: {
+      organizationName: string;
+      userName: string;
+      accessToken: Nullable<string>;
+    }) => {
+      if (!accessToken) {
+        return Promise.reject(new Error("AccessToken not provided"));
+      }
+
+      await leaveOrganizationMutation({
+        organizationName,
+        userName,
+        accessToken,
+      });
+
+      return Promise.resolve(organizationName);
+    }
+  );
+};
```

Comment:
@iamnamananand996 Because we cached all the membership data under the organization so you also need to filter out the member whom leave the org

`["organizations", organizationName, "memberships", userName]`

Please scan all the action that did similar thing to this cache key